libraries/AP_Compass/AP_Compass_AK8963.cpp

```diff
@@ -189,18 +167,19 @@ bool AP_Compass_AK8963::init()
 
     /* register the compass instance in the frontend */
     _compass_instance = register_compass();
-    set_dev_id(_compass_instance, AP_COMPASS_TYPE_AK8963_MPU9250);
+
+    set_dev_id(_compass_instance, _bus->get_dev_id());
```

Comment:
I think it's weird to ask the "bus" what's the dev id... maybe this field is in the wrong class? For now what you could do for an easy transition is to let the underlying bus to set the dev id in AP_Compass_AK8963. Later we can check if there's a better way.